File: tests/oop/oop.cpp
#include <iostream>
#include <string>

class tcounter {
public:
    int value;
    tcounter(int v);
    void increment();
    int getvalue();
};

tcounter::tcounter(int v) {
    value = v;
}

void tcounter::increment() {
    value = (value + 1);
}

int tcounter::getvalue() {
    return value;
}

int main() {
    tcounter* c;
    
    c = new tcounter(0);
    c->increment();
    c->increment();
    c->increment();
    std::cout << c->getvalue() << std::endl;
    return 0;
}

File: tests/oop/oop.pas
program OopTest;

type
  TCounter = class
    Value: integer;
    constructor Create(v: integer);
    procedure Increment;
    function GetValue: integer;
  end;

var
  c: TCounter;

constructor TCounter.Create(v: integer);
begin
  Value := v
end;

procedure TCounter.Increment;
begin
  Value := Value + 1
end;

function TCounter.GetValue: integer;
begin
  GetValue := Value
end;

begin
  c := TCounter.Create(0);
  c.Increment;
  c.Increment;
  c.Increment;
  writeln(c.GetValue())
end.
